packages/snap-client/src/Client/transforms/searchResponse.ts

```diff
@@ -214,7 +214,7 @@ transformSearchResponse.result = (rawResult: rawResult): SearchResponseModelResu
 	const attributes = Object.keys(rawResult)
 		.filter((k) => CORE_FIELDS.indexOf(k) == -1)
 		// remove 'badges' from attributes - but only if it is an object
-		.filter((k) => !(k == 'badges' && typeof rawResult[k] == 'object' && !Array.isArray(rawResult[k])))
+		.filter((k) => !(k == 'badges' && typeof rawResult[k] == 'object' && Array.isArray(rawResult[k]) && typeof rawResult[k]?.[0] == 'object'))
```

Comment:
Seems like the logic for checking `typeof rawResult[k] == 'object'` is unnecessary, as the next check is if it is an array.